files/default/neutron-ha-tool.py

```diff
@@ -598,15 +598,14 @@ def migrate_l3_routers_from_agent(qclient, agent, targets, agent_picker,
                                   router_filter, noop, wait_for_router,
                                   delete_namespace):
     LOG.info("Querying agent_id=%s for routers to migrate away", agent['id'])
-    router_id_list = list_routers_on_l3_agent(qclient, agent['id'])
-    router_id_list = router_filter.filter_routers(router_id_list)
```

Comment:
It seems this line has gone fishing. The aim of the router filter is to be able to only migrate a specified set of routers - if we remove that, we might be in trouble as user would expect only some routers are migrated, and all of the routers will be. The other question - why it was not caught by tests - might be valid from your side. That's something that needs to be fixed.